static/app/views/issueList/customViewsHeader.tsx

```diff
@@ -175,39 +176,81 @@ function CustomViewsIssueListHeaderTabsContent({
         },
         pathname: `/organizations/${organization.slug}/issues/`,
       });
+      tabListState?.setSelectedKey(draggableTabs[0].key);
       return;
     }
     // if a viewId is present, check if it exists in the existing views.
     if (viewId) {
       const selectedTab = draggableTabs.find(tab => tab.id === viewId);
-      if (
-        selectedTab &&
-        (query !== selectedTab!.query || sort !== selectedTab!.querySort)
-      ) {
+      if (selectedTab && query && sort) {
         // if a viewId exists but the query and sort are not what we expected, set them as unsaved changes
-        setDraggableTabs(
-          draggableTabs.map(tab =>
-            tab.key === selectedTab!.key
-              ? {
-                  ...tab,
-                  unsavedChanges: [query, sort],
-                }
-              : tab
-          )
-        );
-      } else if (!selectedTab) {
+        const isCurrentQuerySortDifferentFromExistingUnsavedChanges =
+          selectedTab.unsavedChanges &&
+          (selectedTab.unsavedChanges[0] !== query ||
+            selectedTab.unsavedChanges[1] !== sort);
+
+        const isCurrentQuerySortDifferentFromSelectedTabQuerySort =
+          query !== selectedTab.query || sort !== selectedTab.querySort;
```

Comment:
I personally think verbose, XXL variable names are the lesser evil compared to outrageous boolean algebra, but I'd accept if we'd prefer to remove these.